apps/safety/main.cpp

```diff
@@ -1,13 +1,107 @@
 #include "mbed.h"
+#include "CANMsg.h"
+#include "INA_226.h"
+#include "thermistor.h"
+#include "rover_config.h"
 
-DigitalOut led1(LED1);
+// Datasheet for INA226 http://www.ti.com/lit/ds/symlink/ina226.pdf
+// current sensor data https://docs.google.com/spreadsheets/d/1Qn4QbEJ1Ia54vYnTgaTvLxHralQvCz6PEGaSF_O1L20/edit#gid=0
 
-const int k_interval_ms = 500;
+//Serial communications for debugging
+Serial pc(SERIAL_TX, SERIAL_RX, ROVER_DEFAULT_SERIAL_BAUD_RATE);
+
+//CAN initialization 
+CAN can(CAN1_RX, CAN1_TX, ROVER_CANBUS_FREQUENCY); // dont know the actual CAN id's
+CANMsg rxMsg;
+//needs a specific CANID to update values
+constexpr unsigned int EXPECTED_CAN_ID = 0x000;
+
+//measurement thresholds, needs updating, comparison in celcius now
+constexpr float CURRENT_UPPER_LIM = 1;
+constexpr float CURRENT_LOWER_LIM = 0;
+constexpr float CURVOLTAGE_UPPER_LIM = 1;
+constexpr float VOLTAGE_LOWER_LIM = 0;
+constexpr float TEMP_UPPER_LIM = 1;
+constexpr float TEMP_LOWER_LIM = 0;    
+
+//sample struct for component specific data
+struct elbow_data
```

Comment:
call it elbow_config for consistency for this type of struct. Look at the actuator class for reference. It's prob also better to accept this struct in the constructor of the current sensor class